adapters/adyoulike/adyoulike_test.go

```diff
@@ -0,0 +1,113 @@
+package adyoulike
+
+import (
+	"encoding/json"
+	"net/http"
+	"testing"
+
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/adapters/adapterstest"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/stretchr/testify/assert"
+)
+
+const testsBidderEndpoint = "https://localhost/bid/4"
+
+func TestJsonSamples(t *testing.T) {
+	bidder, buildErr := Builder(openrtb_ext.BidderAdyoulike, config.Adapter{
+		Endpoint: testsBidderEndpoint})
+
+	if buildErr != nil {
+		t.Fatalf("Builder returned unexpected error %v", buildErr)
+	}
+
+	adapterstest.RunJSONBidderTest(t, "adyouliketest", bidder)
+}
+
+func TestMakeRequestInvalidParams(t *testing.T) {
+	bidder, buildErr := Builder(openrtb_ext.BidderAdyoulike, config.Adapter{
+		Endpoint: testsBidderEndpoint})
+
+	assert.Nil(t, buildErr, "buildErr must be nil")
+
+	var reqInfo adapters.ExtraRequestInfo
+	var req openrtb.BidRequest
+	req.ID = "test_id"
+
+	reqExt := `{"prebid":{}}`
+	impExt := `{"bidder":{"placementId":123}}`
+	req.Ext = []byte(reqExt)
+
+	req.Imp = append(req.Imp, openrtb.Imp{ID: "1_0", Ext: []byte(impExt)})
+
+	bids, errs := bidder.MakeRequests(&req, &reqInfo)
+
+	assert.EqualError(t, errs[0], "Key path not found")
+	assert.Len(t, bids, 0)
+}
+
+func TestMakeRequestTagId(t *testing.T) {
+
+	bidder, buildErr := Builder(openrtb_ext.BidderAdyoulike, config.Adapter{
+		Endpoint: testsBidderEndpoint})
+
+	assert.Nil(t, buildErr, "buildErr must be nil")
+
+	var reqInfo adapters.ExtraRequestInfo
+	var req openrtb.BidRequest
+	req.ID = "test_id"
+
+	reqExt := `{"prebid":{}}`
+	impExt1 := `{"bidder":{"placement":"placementid1"}}`
+	impExt2 := `{"bidder":{"placement":"placementid2"}}`
+	req.Ext = []byte(reqExt)
+
+	req.Imp = append(req.Imp, openrtb.Imp{ID: "1_0", Ext: []byte(impExt1)})
+	req.Imp = append(req.Imp, openrtb.Imp{ID: "1_1", Ext: []byte(impExt2)})
+
+	requests, errs := bidder.MakeRequests(&req, &reqInfo)
+
+	assert.Len(t, errs, 0)
+
+	var request *openrtb.BidRequest
+	json.Unmarshal(requests[0].Body, &request)
+
+	for _, imp := range request.Imp {
+
+		assert.True(t, imp.ID == "1_0" || imp.ID == "1_1")
+
+		if imp.ID == "1_0" {
+			assert.Equal(t, imp.TagID, "placementid1")
+		} else if imp.ID == "1_1" {
+			assert.Equal(t, imp.TagID, "placementid2")
+		}
+	}
+}
+
+func TestOpenRTBSurpriseResponse(t *testing.T) {
```

Comment:
Please consider converting this to JSON tests. See other adapter tests in their supplemental test folder for guidance (e.g. `/adapters/conversant/conversanttest/supplemental/server_badresponse.json`).

Reply:
Can we have multiple error test case in one json file or does that mean create one json file for each error case ?
This short method covers four error cases. I don't see the point if we have to create four files for the same cases ?

Reply:
We find it more manageable and simple if we have one JSON test case per error scenario. You can have multiple error testcases in the `adapters/adyoulike/adyouliketest/supplemental/` directory and `TestJsonSamples` will run them all making it unnecessary to have `TestMakeRequestInvalidParams(t *testing.T)`, `TestMakeRequestTagId(t *testing.T` and TestOpenRTBSurpriseResponse(t *testing.T)` at all. File `adapters/adyoulike/adyoulike_test.go` should only have the `TestJsonSamples(t *testing.T)` test. Please taka look at our documentation or other adapter's test or feel free to hit us with any questions you may have.
```
</prebid/prebid-server/
▾ adapters/
  ▾ adyoulike/
    ▾ adyouliketest/
      ▸ exemplary/
      ▸ params/
      ▾ supplemental/
          invalid-tagID.json
          wrong-impression-ext.json
            .
            .
      adyoulike.go
      adyoulike_test.go
      params_test.go
      usersync.go
      usersync_test.go
```